### DISCUSSION TITLE
Bind port 80 permission denied

### DISCUSSION THREAD

**Discussion by codycrypto** [smallstep/certificates#1242] [Q&A] [answered]
Trying to complete an ACME request through step-cli

Receiving this error - "ListenAndServe(): listen tcp :80: bind: permission denied"

Tried running with sudo and this is what is returned "flag '--ca-url' is required unless the '--token' flag is provided"

I have a private local CA configured with step-ca.

**Comment by maraino** [comment, marked as answer]
Hi @codycrypto, the error happens because step/step-cli does not have permission to listen on port 80, required by the http-01 challenge. 

There are different solutions for that. If you're in a Linux or a macOS, you can run it with `sudo`. Alternatively, on a Linux, you can also add capabilities to the binary, a command like:

```
sudo setcap CAP_NET_BIND_SERVICE=+eip path/to/step
```

On windows, I don't really know, running that as administrator?

The problem can also be that there's another program running on port 80.

**Reply by codycrypto** [reply]
Thank you! I think I found that same command digging around stack overflow at the same time you replied....that solved the issue.  Thanks again.
